packages/aws-cdk-lib/aws-batch/lib/job-queue.ts

```diff
@@ -209,6 +297,34 @@ export class JobQueue extends Resource implements IJobQueue {
       order,
     });
   }
+
+  private renderJobStateTimeLimitActions(
+    jobStateTimeLimitActions?: JobStateTimeLimitAction[],
+  ): CfnJobQueue.JobStateTimeLimitActionProperty[] | undefined {
+    if (!jobStateTimeLimitActions || jobStateTimeLimitActions.length === 0) {
+      return;
+    }
+
+    return jobStateTimeLimitActions.map((action, index) => renderJobStateTimeLimitAction(action, index));
+
+    function renderJobStateTimeLimitAction(
+      jobStateTimeLimitAction: JobStateTimeLimitAction,
+      index: number,
+    ): CfnJobQueue.JobStateTimeLimitActionProperty {
+      const maxTimeSeconds = jobStateTimeLimitAction.maxTime.toSeconds();
+
+      if (maxTimeSeconds < 600 || maxTimeSeconds > 86400) {
+        throw new Error(`maxTime must be between 600 and 86400 seconds, got ${maxTimeSeconds} seconds at jobStateTimeLimitActions[${index}]`);
+      }
+
+      return {
+        action: jobStateTimeLimitAction.action ?? JobStateTimeLimitActionsAction.CANCEL,
+        maxTimeSeconds,
+        reason: jobStateTimeLimitAction.reason,
```

Comment:
`Reason` is also a required prop, we should check for an empty reason.

Reply:
@GavinZZ 
My understanding is that you would like to add validation around the Reason property not being undefined, is that correct?
Since Reason is a required property, it would show an error in the IDE etc. if it was undefined. 
I don't quite understand the reasoning behind needing to add the above validation, so could you please explain the intention behind it?

Reply:
Right my bad, I wasn't in my right mind when I put this comment. Please disregard.